### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Ibrahim-Khan88**

### CODE REVIEW

File: src/main/java/com/logni/credit/service/controller/LoanAppliactionController.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      return ResponseEntity.ok().body(loanApplicationRepository.findAll());
+   }
+
+   @GetMapping("/document/{fileName}")
+   public ResponseEntity getEventImage(@PathVariable("fileName") String fileName, HttpServletRequest request) {
+      Resource resource = fileStorageService.loadFileAsResource(fileName);
+      String contentType = null;
+      try {
+         contentType = request.getServletContext().getMimeType(resource.getFile().getAbsolutePath());
+      } catch (IOException ex) {
```
Comment (azomshahriar) [inline src/main/java/com/logni/credit/service/controller/LoanAppliactionController.java:63]: Plz log the exception

File: src/main/java/com/logni/credit/service/controller/LoanController.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            loanCreate.get().setLoanStatus(loanReceived.getLoanStatus());
+            loanRepository.save(loanCreate.get());
+            return ResponseEntity.status(HttpStatus.CREATED).body("Successfully Updated");
+         }
+         return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Loan Is Rejected");
+      }
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Loan Not Found");
+   }
+
+   // todo update loan status
```
Comment (azomshahriar) [inline src/main/java/com/logni/credit/service/controller/LoanController.java]: remove comment if done

File: src/main/java/com/logni/credit/service/controller/LoanController.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      if (loan != null) {
+         return ResponseEntity.status(HttpStatus.CREATED).body(loan);
+      } else {
+         return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
+      }
+   }
+
+   @PostMapping("/approved")
+   ResponseEntity<String> updateLoan(@RequestBody LoanApproved loanReceived) {
+      Optional<Loan> loanCreate = loanRepository.findById(loanReceived.getLoanId());
```
Comment (azomshahriar) [inline src/main/java/com/logni/credit/service/controller/LoanController.java]: loanCreate -> createdLoan

File: src/main/java/com/logni/credit/service/controller/LoanProductController.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+@RestController()
+@RequestMapping("/api/product")
+@Slf4j
+public class LoanProductController {
+	@Autowired
+	private LoanProductRepository loanProductRepository;
+
+	@SuppressWarnings("unchecked")
+	@PostMapping()
```
Comment (azomshahriar) [inline src/main/java/com/logni/credit/service/controller/LoanProductController.java]: remove ()

File: src/main/java/com/logni/credit/service/controller/LoanProductController.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+			return ResponseEntity.ok(loanProductOptional.get());
+		} else {
+			return ResponseEntity.notFound().build();
+		}
+	}
+
+	@DeleteMapping("/{id}")
+	public ResponseEntity<?> deleteLoanProductId(@PathVariable(value = "id") int loanProductId) {
+		boolean exist = loanProductRepository.existsById(loanProductId);
+		if (exist) {
```
Comment (azomshahriar) [inline src/main/java/com/logni/credit/service/controller/LoanProductController.java]: loanProductRepository.findById(loanProductId).ifPresent(loanProductRepository::delete);

Comment (azomshahriar) [reply src/main/java/com/logni/credit/service/controller/LoanProductController.java]: single line code

File: src/main/java/com/logni/credit/service/service/FileStorageService.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    public Resource loadFileAsResource(String fileName) {
+        try {
+            Path filePath = Paths.get(uploadFolder, fileName);
+            log.info(filePath.toString());
+            log.info(filePath.toUri().toString());
+            Resource resource = new UrlResource(filePath.toUri());
+            if (resource.exists()) {
+                return resource;
+            } else {
+                throw new RuntimeException("File not found " + fileName);
```
Comment (azomshahriar) [inline src/main/java/com/logni/credit/service/service/FileStorageService.java]: LoofiRuntimeException??

**Review by azomshahriar — CHANGES_REQUESTED** [review]

File: src/main/java/com/logni/credit/service/service/FileStorageService.java (hunk 6 above)
Comment (Ibrahim-Khan88) [reply src/main/java/com/logni/credit/service/service/FileStorageService.java]: Resolved all issues

**Pull request merged**